torchci/lib/bot/retryBot.ts

```diff
@@ -0,0 +1,74 @@
+import { Probot } from "probot";
+
+const FAILURE_CONCLUSIONS = ["failure", "cancelled", "timed_out"];
+
+function retryBot(app: Probot): void {
+  app.on("workflow_run.completed", async (ctx) => {
+    const workflowName = ctx.payload.workflow_run.name;
+    const attemptNumber = ctx.payload.workflow_run.run_attempt;
+    if (
+      ctx.payload.workflow_run.conclusion === "success" ||
+      ctx.payload.workflow_run.head_branch !== "master" ||
+      attemptNumber > 1 ||
+      !["lint", "pull"].includes(workflowName)
+    ) {
+      return;
+    }
+    const owner = ctx.payload.repository.owner.login;
+    const repo = ctx.payload.repository.name;
+    const runId = ctx.payload.workflow_run.id;
+    const workflowJobs =
+      await ctx.octokit.rest.actions.listJobsForWorkflowRunAttempt({
+        owner,
+        repo,
+        run_id: runId,
+        attempt_number: attemptNumber,
+      });
+
+    const failedJobs = workflowJobs.data.jobs.filter((job) =>
+      FAILURE_CONCLUSIONS.includes(job.conclusion!)
+    );
+    if (failedJobs.length > 5) {
+      // if you have more than 5 failing jobs, its probably either a real failure, a landrace,
+      // or a widespread outage that wouldn't be helped by retries
+      return;
+    }
+    const shouldRetry = failedJobs.filter((job) => {
+      // always rerun lint
+      if (workflowName === "lint") {
```

Comment:
Is a lint special case really needed? Or will this code not work the same without it?

Reply:
i put this here to make sure that lint is always rerun if anything fails.  The filter later on could potentially remove lint jobs on accident because it's pretty broad